src/main/scala/org/broadinstitute/dsde/workbench/avram/util/AvramResult.scala

```diff
@@ -0,0 +1,30 @@
+package org.broadinstitute.dsde.workbench.avram.util
+
+import cats.data.EitherT
+import cats.effect.IO
+
+// Note: I'm unsatisfied with AvramResult as a name but I can't think of anything better. Suggestions welcome. -breilly
+object AvramResult {
+  /**
+    * Convenience type that effectively represents IO[Either[ErrorResponse, A]]
+    */
+  type AvramResult[A] = EitherT[IO, ErrorResponse, A]
+
+  /**
+    * Promote an IO to an AvramResult.
+    */
+  def AvramResult[A](io: IO[A]): AvramResult[A] = EitherT.right(io)
+
+  /**
+    * Promote an Either to an AvramResult.
+    */
+  def AvramResult[A](either: Either[ErrorResponse, A]): AvramResult[A] =
```

Comment:
These `AvramResult` functions allow for `IO`-like usage:
```
AvramResult { ??? }
```
which seemed convenient. However, I don't entirely like it because it's not always clear which of these functions is being called. On the flip side, it's nice to not have to worry about which is being called.

_However_, I had to stop short of adding what I originally thought would be convenient:
```
def AvramResult[A](a: A): AvramResult[A] = EitherT.pure(a)
```
The problem with this is that you might trip over it when you really mean to be calling the version that takes an `IO`. To illustrate this, add the above method and then change `HttpSamDao.queryAction` to use `sttpBackend` instead of `catsSttpBackend`. It still compiles _and_ the tests still pass! However, even though everything is still lexically "wrapped" in `AvramResult`, it's no longer safely in an `IO`... and I can't think of a test that would expose this. Almost makes me want to add this:
```
def AvramResult[A](a: A): AvramResult[A] = ??? // something here that causes compilation failure if anything references it
```
@rtitle, do you have any ideas about how to make this better?

Reply:
`AvramResult` is just a type alias for `EitherT` - why not just make callers call `pure` when passing a pure value? This may be why `EitherT` doesn't have overloaded methods -- they make you explicitly call `.fromEither`, `.pure`, `.right` depending on what type you have. Makes it harder to "lift" the wrong type by mistake.

Also instead of `def AvramResult` you can probably do `def apply` and still get the `AvramResult { ??? }` syntax since you're already inside an `AvramResult` object.

Reply:
`apply` doesn't seem to work... I think maybe because it's an `object` rather than a `class` or `trait`? I'm not sure. I had the same initial thought but wasn't able to get any use of `apply` to compile. I think I'd have to promote `AvramResult` to a full type to get that.

Although `AvramResult` is a type alias, it doesn't seem to get the type's functions. For example, I can't just call `AvramResult.right()`. I'm sure there's a good reason why that's the case but I'm not entirely surprised either.

Provided I put the `AvramResult` type alias in a package object, I was able to add these functions to the `AvramResult` object:
* `pure` - proxies to `EitherT.pure`
* `fromEither` - proxies to `EitherT.fromEither`
* `fromIO` - proxies to `EitherT.right`
* `fromError` - uses `EitherT.left` to make a pure-like function for errors

I think I like it. Code now reads like:
```
AvramResult.pure(theAnswer)
AvramResult.fromEither(doSomethingThatMightFail())
AvramResult.right(IO(doSomethingEffecty()))
AvramResult.fromError(ErrorResponse(500, "Boom!"))
```
What do you think of that?

Reply:
I went ahead and pushed this. Let me know what you think. It's easy enough to undo.